htlcswitch/link.go

```diff
@@ -874,6 +879,22 @@ func (l *channelLink) htlcManager() {
 			// what they sent us before.
 			// TODO(halseth): ban peer?
 			case err == lnwallet.ErrInvalidLocalUnrevokedCommitPoint:
+				forceClose, err := l.channel.ForceClose()
+				if err != nil {
+					log.Errorf("Unable to force close "+
+						"channel: %v", err)
+					return
+				}
+
+				err = l.channel.MarkCommitmentBroadcasted(
```

Comment:
Do we need to do all of this now given that we fail the link and signal that it should be force closed below? So it'll do all of this, but at the contract court level.

Reply:
Yeah, within `handleLinkFailure` (peer.go) we handle all the clean up. As a result, we only need to modify this closure: https://github.com/lightningnetwork/lnd/blob/master/contractcourt/chain_arbitrator.go#L234

Recall that care was taken to ensure that no link is active, and no new link can be loaded in while we're carrying out the force close, to avoid an accidental self breach. We may need to modify the ordering here slightly,  to ensure that we're able to carry out the restart properly. Otherwise, if we crash after that fist call, then the channel is marked as borked, so it won't be resumed by the chain arb.

Reply:
Removed and added comment